packages/assets-controllers/src/TokenRatesController.ts

```diff
@@ -447,14 +483,21 @@ export class TokenRatesController extends StaticIntervalPollingControllerV1<
     tokenAddresses: Hex[];
     chainId: Hex;
     nativeCurrency: string;
-  }): Promise<ContractExchangeRates> {
+  }): Promise<ContractInformations> {
     if (!this.#tokenPricesService.validateChainIdSupported(chainId)) {
-      return tokenAddresses.reduce((obj, tokenAddress) => {
-        return {
-          ...obj,
-          [tokenAddress]: undefined,
-        };
-      }, {});
+      return tokenAddresses.reduce(
+        (obj, tokenAddress) => {
+          obj.marketData = {
+            ...obj.marketData,
+            [tokenAddress]: undefined,
+          };
+
+          return obj;
```

Comment:
I would look into using the delete keyword instead of setting undefined explicitely. We should avoid setting undefined explicitely.

proposal:

```
 const newMarketData = { ...obj.marketData };
  delete newMarketData[tokenAddress];
  return {
    ...obj,
    marketData: newMarketData,
  };
```